File: code/euler1.c
#include <stdio.h>

int main(int argc, char *argv[]) {
    if(argc != 4) {
        fprintf(stderr,"No no no... %d \n", argc);
        return(-1);
    }
    int a = atoi(argv[1]);
    int b = atoi(argv[2]);
    int limit = atoi(argv[3]);
    
    
    int i = 0;
    int sum = 0;
    for(i = 0; i<limit;i++) {
        if(i % a == 0 || i % b == 0 ){
            sum +=i;
        }
    }
    
    printf("%d\n",sum);
    return(0);
}

File: code/euler1.go
package main

import "fmt"

func multipleSum(x int, y int, limit int)  int {
	sum := 0
	for i := 0;  i < limit; i++ {
		if i % x == 0 || i % y == 0 {
			sum += i
		}
	}
	return sum
}

func main() {
	sum := multipleSum(3,5,1000)
	fmt.Println(sum)
}
